src/_P076_HLW8012.ino

```diff
@@ -128,24 +143,53 @@ boolean Plugin_076(uint8_t function, struct EventStruct *event, String& string)
     }
 
     case PLUGIN_GET_DEVICEVALUENAMES: {
-      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0],
-               PSTR(PLUGIN_VALUENAME1_076));
-      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[1],
-               PSTR(PLUGIN_VALUENAME2_076));
-      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[2],
-               PSTR(PLUGIN_VALUENAME3_076));
-      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[3],
-               PSTR(PLUGIN_VALUENAME4_076));
+      for (uint8_t i = 0; i < VARS_PER_TASK; ++i) {
+        if (i < P076_NR_OUTPUT_VALUES) {
+          uint8_t choice = PCONFIG(i + P076_QUERY1_CONFIG_POS);
+          ExtraTaskSettings.setTaskDeviceValueName(i, p076_getQueryString(choice));
+        } else {
+          ExtraTaskSettings.clearTaskDeviceValueName(i);
+        }
+      }
       break;
     }
 
+    case PLUGIN_SET_DEFAULTS: {
```

Comment:
This is only called when adding the task.
What about existing configurations?
Existing configurations will have P076_QUERY1...4 all equal 0.
So maybe during the PLUGIN_INIT call you could check to see if all are set to 0 and then apply these defaults.

Reply:
> Is the code under PLUGIN_SET_DEFAULTS not only called when I add a new task?

Only when adding a new task, not when starting a task which was already configured before installing the build with your changes.

Reply:
Okay,
I added an commit to set the defaults also in PLUGIN_INIT if they are note set.